src/main/java/com/questhelper/helpers/quests/thecurseofarrav/MetalDoorSolver.java

```diff
@@ -0,0 +1,328 @@
+package com.questhelper.helpers.quests.thecurseofarrav;
+
+import com.questhelper.requirements.item.ItemRequirement;
+import com.questhelper.requirements.widget.WidgetModelRequirement;
+import com.questhelper.requirements.widget.WidgetPresenceRequirement;
+import com.questhelper.requirements.widget.WidgetTextRequirement;
+import com.questhelper.steps.ConditionalStep;
+import com.questhelper.steps.DetailedOwnerStep;
+import com.questhelper.steps.DetailedQuestStep;
+import com.questhelper.steps.ItemStep;
+import com.questhelper.steps.ObjectStep;
+import com.questhelper.steps.QuestStep;
+import java.util.List;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+import com.questhelper.steps.WidgetStep;
+import com.questhelper.steps.widget.WidgetDetails;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.ItemID;
+import net.runelite.api.ObjectID;
+import net.runelite.api.annotations.Interface;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.events.GameTick;
+import net.runelite.client.eventbus.Subscribe;
+import static com.questhelper.requirements.util.LogicHelper.and;
+import static com.questhelper.requirements.util.LogicHelper.not;
+
+@Slf4j
+public class MetalDoorSolver extends DetailedOwnerStep
+{
+	private static final int STRIP_1_INDEX = 2;
+	private static final int STRIP_2_INDEX = 1;
+	private static final int STRIP_3_INDEX = 3;
+	private static final int STRIP_4_INDEX = 0;
+	/**
+	 * Maps the relevant numbers from the Metal door puzzle to their corresponding rows.
+	 * The first column is for strip 4
+	 * The second column is for strip 2
+	 * The third column is for strip 1
+	 * The fourth column is for strip 3
+	 */
+	private static final int[][] CODE_MAPPING = new int[][]{
```

Comment:
Would be good for this to have a few extra features:

1. Arrow highlight to have number of times you still need to click it.
2. Overlay saying what the answer is
3. Allow for down arrow to be used if it's closer to answer

The existing door solvers from like Tribal Totem and SOTN should provide templates for those bits.

Reply:
This has been done, but in a bit of a unique way since I I'm using a DetailedOwnerStep instead of a straight up QuestStep